contrib/awslambda/python/src/python/pants/contrib/awslambda/python/rules/targets.py

```diff
@@ -0,0 +1,28 @@
+# Copyright 2020 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+from pants.engine.target import COMMON_TARGET_FIELDS, Dependencies, StringField, Target
+
+
+class PythonAwsLambdaBinaryField(StringField):
+    """Target spec of the `python_binary` that contains the handler."""
+
+    alias = "binary"
```

Comment:
I think this should be required, but Benjy and I couldn't figure out how the V1 code works because it also looks at the `dependencies` field to determine what the binary is..So, I left it as optional.

NB that this binding is solely so that we parse the BUILD file and can pass it down to V1. V1 will still express its "requiredness" itself.